modules/ROOT/pages/http-caching-policy.adoc

```diff
@@ -6,241 +6,419 @@ endif::[]
 
-*Note:* Values persisted in Object Store v2 appear as binary, not in plain text.
 
-== What this Policy Stores
+== How This Policy Works
 
-This policy stores a whole HTTP response from the backend as received by the policy including the status codes, headers, payload, and session variables. If there are policies that edit the response before it is received by this policy, those changes are also present in the response received and therefore stored. If there are policies that make modifications to the response after it passes through the caching policy, those changes won’t be stored, but are replicated each time the value is retrieved from the cache.
+When an HTTP request reaches an endpoint, what happens next depends on whether that request does not already exist in cache (a cache miss) or whether it does (a cache hit).
 
-Each entry has a size limit of 1 MB. In case of HTTP Responses that do not provide content length information, the policy tries to read up to that limit and stores it when it is not reached or continues without storing it, when that limit is exceeded.
+=== Cache Miss
 
-== How this Policy Works
+In a cache miss scenario, the HTTP Caching policy checks whether a response to the submitted request is already cached. The request reaches the backend and the process is initiated only when it's a cache miss. The diagram illustrates the flow of events that occur in this scenario:
 
-When the policy receives a request, it calculates a key under which to store the response. When a second request with the same key arrives, the policy searches the cache for the previously stored response and returns it. If a matching response is found in the cache, the request does not reach either the policies that are defined after the cached one or the API implementation or proxy.
+image:http-policy-cache-miss.png[]
 
-== Where Entries are Stored
+. The client sends a request to the API.
+//Note that this step refers to the actors as "the client" and "the API" rather than user request, HTTP request, endpoint, or backend. Using the same terms for the same things throughout a narrative eases comprehension.
+. The request proceeds through other policies in the policy chain before reaching the HTTP Caching policy.
+. The HTTP Caching policy checks if the request is already in the object store and, if not, the cache miss process is initiated.
+. The request continues throughout the chain of other remaining policies.
+. The request reaches the backend and completes all required processing.
+. The backend responds and the response travels back through the policy chain.
+. The response reaches the HTTP Caching policy. 
+// (@lorena--the request reaches the policy or the response?)
+. Caching hits the object store again and attempts to save the <<key,key>> of the request for later reuse.
+//@Lorena--Do we mean that the policy hit the object store again?
```

Comment:
@priyaayyar Yes, exactly, the policy hits the object store.

Reply:
Done! Thanks!